pkg/cmd/infra/deployer/deployer.go

```diff
@@ -196,5 +203,53 @@ func (d *Deployer) Deploy(namespace, deploymentName string) error {
 	} else {
 		glog.Infof("Deploying from %s to %s (replicas: %d)", deployutil.LabelForDeployment(from), deployutil.LabelForDeployment(to), desiredReplicas)
 	}
-	return strategy.Deploy(from, to, desiredReplicas)
+	if err := strategy.Deploy(from, to, desiredReplicas); err != nil {
+		return err
+	}
+	return d.after(config, to)
```

Comment:
> basically, if I want to fail a deployment because of a post hook, I should be allowed to.

Agreed. This means hooks/after should hold an option for this.